kubelet/api: split RuntimeService interface

tmrts opened the pull request:
Splits `RuntimeService` interface into smaller interfaces
to make testing easier and delineate the responsibilities.

Its a non-breaking change to the previous users of `api.RuntimeService`

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30619)

<!-- Reviewable:end -->

yujuhong commented on pkg/kubelet/api/services.go:
```diff
@@ -23,13 +23,44 @@ import (
 	// The definition of PodSandbox is at https://github.com/kubernetes/kubernetes/pull/25899
 	CreatePodSandbox(config *runtimeApi.PodSandboxConfig) (string, error)
+	// CreateContainer creates a new container in specified PodSandbox.
+	CreateContainer(podSandboxID string, config *runtimeApi.ContainerConfig, sandboxConfig *runtimeApi.PodSandboxConfig) (string, error)
```
Hm...I think this belongs to `ContainerManager`.

tmrts replied:
The reason I put it there was its clear that in order to create a container, information about PodSandboxes has to be passed and there has to be some logic that makes sense of all the PodSandbox configuration. So, this way data is close to the methods that operate on the data. 

If its not a convincing enough reason, I can just add it to ContainerManager.

yujuhong replied:
The reason is valid, but I still think the method should be included in the ContainerManager instead, since it's part of the container lifecycle management. We can change it later if it becomes obvious in the tests/code that this belongs to the sandbox manager.

tmrts replied:
Fair enough :+1: moved the method, PTAL

tmrts committed:
kubelet/api: extract RuntimeVersioner interface

tmrts committed:
kubelet/api: extract ContainerManager interface

tmrts committed:
kubelet/api: extract PodSandboxManager interface

Splits `RuntimeService` interface into smaller interfaces
to make testing easier and to delineate responsibilities.

yujuhong commented:
LGTM. Thanks!
